### PR TITLE
Update legacy php to v10 + new proto

### PR DESCRIPTION
**Pull request by tbillington**

### CODE REVIEW

File: src/Sajari/Record/Value.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    /**
+     * @return ValuesEntry
+     */
+    public function Proto()
+    {
+        $actualValue = new EngineValue();
+
+        if (is_array($this->value)) {
```
Comment (dhowden) [inline src/Sajari/Record/Value.php:65]: This should be using the generic value code.

File: src/Sajari/Record/Key.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+     * @return EngineKey
+     */
+    public function Proto()
+    {
+        $protoKey = new EngineKey();
+        $protoKey->setField($this->field);
+
+        $value = new EngineValue();
+        if (is_array($this->value)) {
+          $repeated = new EngineValueRepeated();
```
Comment (dhowden) [inline src/Sajari/Record/Key.php:52]: Generic value.

File: src/Sajari/Search/FieldFilter.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    /**
+     * @return EngineFilterField
+     */
+    public function Proto()
+    {
+        $ff = new EngineFilterField();
+        $ff->setField($this->field);
+
+        $value = new Value();
```
Comment (dhowden) [inline src/Sajari/Search/FieldFilter.php:105]: Generic value.

**Review by dhowden — CHANGES_REQUESTED** [review]

File: src/Sajari/Record/Value.php (hunk 1 above)
Comment (tbillington) [reply src/Sajari/Record/Value.php:65]: This one is actually going in the other direction 😛

File: src/Sajari/Record/Key.php (hunk 2 above)
Comment (tbillington) [reply src/Sajari/Record/Key.php:52]: This code is in `Value->Proto()`, I'll clean that up and use that.

File: src/Sajari/Search/FieldFilter.php (hunk 3 above)
Comment (tbillington) [reply src/Sajari/Search/FieldFilter.php:105]: Same as above.

**Pull request merged**